### PR TITLE
Mini-Hypospray

### PR DESCRIPTION
**Pull request by Steelpoint**
This PR adds a new, expensive, medical item to the Medical Vendors for Doctors to purchase of their own funds.

hypo

The mini-hypospray functions similarly to the standard Hypospray the CMO is issued with. However the mini variant can only hold up to 10 units at any time, down from the usual 30. The mini-hypospray costs $1200 to purchase from a vendor, each vendor stocks two each.

:cl: Steelpoint
add: The DeForest Medical Corporation has announced a new line of mini-hyposprays, created with cheaper materials and holding less chemicals than the standard hypospray, thus easier to produce.
add: Nanotrasen does not consider the new mini-hyposprays to be standard issue. So interested medical personnel will need to buy one from their Medical Vendor from their own pay.
/:cl:

### CODE REVIEW

**Comment by improvedname** [comment]
nuerotoxin validhunting edition

**Comment by optimumtact** [comment]
so instead of trying to come up with something unique you just make the CMO's tool less unique?

**Comment by Steelpoint** [comment]
To be fair that has already occured with the prevalence of the Cyborg Hyposprays.

If the issue is trying to keep head of staff items solely unique instead of something that can be bought or researched, then we have a major issue since the majority of head items are easily acquired by crew members. The CE's tools can be mass produced by anyone with Sci or Engi access. The CMO's compact defib is another researched tool, as is his adv analyzer. The HoS's Tele Shield is unlocked very early in the tech tree. Etc.

I figured this would give doctors something to save up for and something for them to sink funds into. If lethality is something of interest they can just nab a syringe gun and get a bluespace or diamond tipped syringe.

**Comment by optimumtact** [comment]
I suppose yeah, it's just frustrating to try to add meaningful differences to jobs and roles and then people come along and reuse the item for their own stuff.

like the webbing pr was a good idea, and I know it's hard to come up with generally unique stuff but still you know :cry:

**Comment by Steelpoint** [comment]
One concept I can think of is reusable autoinjector that lets you swap out injector caps that can be filled with a chem of choice, or to prevent "fun" you have to buy pre-made caps that come pre-filled with chemicals. 

But I'm not a coder so I can only work within a very narrow area.

**Comment by nicbn** [comment]
Rework the sprite, this doesn't look any creative

**Comment by IllotAFV** [comment]
I thought,that Syndicate needs instant injection thingy,not omnizine or high volume container. But who cares about that lore anyway.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -76,6 +76,13 @@
 	container_type = NONE
 	list_reagents = list("magillitis" = 5)
 
+/obj/item/reagent_containers/hypospray/mini
+	name = "mini hypospray"
+	desc = "New varient of the standard air-needle autoinjector. Holds less chemicals than the standard hypospray but is easier to produce as a result."
```
Comment (XDTM) [inline code/modules/reagents/reagent_containers/hypospray.dm]: variant

**Comment by lolman360** [comment]
another good steelpoint pr. when will he stop?

**Comment by GuyonBroadway** [comment]
Make it need a 0.5 sec use delay cus its not as easy to use as a full fat hypospray. That way you cant drive by nerotox+perfluro someone.

**Comment by Steelpoint** [comment]
On the lore point, the Syndicate have already demonstrated they have access to autoinjector technology via their Combat Injectors that Nuke Ops can buy and the injector you use to implant a Holoparasite. I think the obj to steal the CMO's hypospray is more intended as a humiliation act or to send a message.

On the spray itself, I'm not sure how to increase a delay, also it only injects 5u at a time. I can always increase the price to make it harder to get if we have concerns of traitor usage, but to be fair traitors have access to far more deadly items.

File: code/modules/reagents/reagent_containers/hypospray.dm (hunk 1 above)
Comment (Steelpoint) [reply code/modules/reagents/reagent_containers/hypospray.dm]: I always have something needing fixing with my PRs.

**Comment by Steelpoint** [comment]
If there are major concerns over traitor or antag usage, we could see to adding a list of chems that can't be used with the hypospray? But I'm not sure how I'd do that.

**Comment by Mickyan** [comment]
I think this is going a bit too far, as people mentioned this is undoubtedly going to be a validhunting tool

As an alternative I'd suggest either a box of epipens with healing chemicals in them or an omnizine hypospray that can't be refilled (obviously wouldn't be as expensive as this one but still expensive enough to be used sparingly)

Something you could also add to the vendor is a compact defib, those can usually be printed at the lathe later in the round but the materials aren't always available

**Comment by zamolxius** [comment]
make it blocked by any clothing at all due to budget needles

**Comment by subject217** [comment]
changelog: "personal" should be "personnel".

also please code more security buffs

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 2]
```diff
@@ -76,6 +76,13 @@
 	container_type = NONE
 	list_reagents = list("magillitis" = 5)
 
+/obj/item/reagent_containers/hypospray/mini
+	name = "mini hypospray"
+	desc = "New variant of the standard air-needle autoinjector. Holds less chemicals than the standard hypospray but is easier to produce as a result."
+	icon_state = "mini_hypo"
+	volume = 10
+	custom_price = 1200
```
Comment (Mickyan) [inline code/modules/reagents/reagent_containers/hypospray.dm]:
```suggestion
	custom_premium_price = 1200
```

Comment (Steelpoint) [reply code/modules/reagents/reagent_containers/hypospray.dm]: Done

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 3]
```diff
@@ -81,7 +81,7 @@
 	desc = "New variant of the standard air-needle autoinjector. Holds less chemicals than the standard hypospray but is easier to produce as a result."
 	icon_state = "mini_hypo"
 	volume = 10
-	custom_price = 1200
+	custom_price_price = 1200
```
Comment (Mickyan) [inline code/modules/reagents/reagent_containers/hypospray.dm]: better luck next time

**Comment by ExcessiveUseOfCobblestone** [comment]
We've gone from security belt with an extra slot to an item previously restricted to just CMOs and traitors? I don't consider borgs since their hypo uses set reagents whereas these allow for any and all. Big note: There are multiple medivends on the map so there can be multiple of these floating around, not including the CMO hypo.

In before 4D chess where this is used to arms race for Sec.

**Comment by Steelpoint** [comment]
This PR is only a proposal, if not taken so I'll simply see if I can something else that'll be fancy to buy. Which I intend to do irrespective of this PRs outcome.

File: code/modules/reagents/reagent_containers/hypospray.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/reagent_containers/hypospray.dm]:
```suggestion
	custom_premium_price = 1200
```

just press accept change lol

**Comment by Wilchenx** [comment]
this should be a tool thats printed in the techfab instead

**Comment by dr-bee** [comment]
Goonstation hyposprays have a reagent whitelist to them to prevent validhunting, consider reducing the price and adding the whitelist to just healing chems. Sure you can kill someone with omnizine but it is significantly harder than hitting them with 5 units of sulfanol and waiting for them to drop.

**Comment by ExcessiveUseOfCobblestone** [comment]
or just not have the hypo instead of maintaining another list

**Comment by MMMiracles** [comment]
I like the idea of the purchasable variant being something that takes one-use capsules you can fill up initially before they pressurize themselves for the hypo to use. That way you can cut down the price somewhat and make the actual cash sink be the little capsules needed to use it.

**Comment by dr-bee** [comment]
Plus the idea of medicine K-cups is hilarious and something an ethically bankrupt ultra-corp like Nanotrasen would do.

**Comment by optimumtact** [comment]
I'm still leaning to not merging this for the reasons I raised above.

**Comment by Steelpoint** [comment]
On further reflection I'm not 100% on this, I can think of a few ideas to make it more balanced (the k-cup idea) but that's beyond my abilities. I'll rethink and see if I can think of a diff idea for Docs to buy (after the adv first aid kit)

**Pull request closed without merging**